### ISSUE TITLE
E1017 Select does not find already supported function when using complex list with nested Selects

### ISSUE THREAD

**Issue by florian0410** [aws-cloudformation/cfn-lint#2895]
### CloudFormation Lint Version

cfn-lint 0.80.4

### What operating system are you using?

Mac

### Describe the bug

When launching a template with complex nested Selects and list to extract value from, it seems to be reporting E1017 while it should not. Templates are correctly deployed and work fine on my side.

Output from command
```
E1017 Select should use a supported function of Fn::FindInMap, Fn::GetAtt, Fn::GetAZs, Fn::If, Fn::Split, Fn::Cidr, Ref
/file1.yml:3189:11
```

### Expected behavior

No E1017 reported by cfn-lint.
Template is working fine in Cloudformation, E1017 should not be reported.

### Reproduction template

AWSTemplateFormatVersion: '2010-09-09'
Description: 'Build EC2 instance'

Resources:
  MountTarget1:
    Type: AWS::EFS::MountTarget
    Properties:
      FileSystemId: fs-1234567svsdabsf76s
# E1017 STARTS HERE
      SubnetId: !Select
        - 0
        - !Select
          - 0
          - [
              [
                "subnet-0987sknlnsdoi9j76",
                "subnet-875jgyjlpzj75j8k0",
                "subnet-5447hnd6hI8js45js"
            ],
            [
                "subnet-0987sknlnsdoi9j76",
                "subnet-875jgyjlpzj75j8k0",
                "subnet-5447hnd6hI8js45js"
            ],
            [
                "subnet-0987sknlnsdoi9j76",
                "subnet-875jgyjlpzj75j8k0",
                "subnet-5447hnd6hI8js45js"
            ] 
          ]
      SecurityGroups: [sg-00qdqeef0a5c345gf]
